components/gitpod-db/src/typeorm/project-db-impl.ts

```diff
@@ -27,4 +28,19 @@ export class ProjectDBImpl implements ProjectDB {
         const repo = await this.getRepo();
         return repo.find({ teamId });
     }
+
+    public async createProject(name: string, cloneUrl: string, teamId: string, appInstallationId: string): Promise<Project> {
+        const repo = await this.getRepo();
+
+        const project: Project = {
+            id: uuidv4(),
+            name,
```

Comment:
Nit: Since `name` is also used as a `slug`, maybe validate it here against a regex? For example like so:

https://github.com/gitpod-io/gitpod/blob/903c8545e5eba0365ba008465ac634b2cfd0c254/components/gitpod-db/src/typeorm/team-db-impl.ts#L74-L76